internal/utils/meta.go

```diff
@@ -62,3 +65,13 @@ func ToggleOwnerReference(on bool, obj, owner metav1.Object, scheme *runtime.Sch
 	}
 	return false, nil
 }
+
+func IsNotFoundWithName(err error, name string) bool {
+	if !k8serrors.IsNotFound(err) {
+		return false
+	}
+	status, ok := err.(k8serrors.APIStatus)
+	return (ok || errors.As(err, &status)) &&
```

Comment:
@leelavg Why the two different approaches? can we always use `errors.As`?

Reply:
there is a possibility that APIStatus could be wrapped and so the interface checks fails but unwrapping it can be successful which provides the `Status()` function satisfying this interface and query the error details.

another thing to note here is, API could just return httpStatusNotFound without setting details of the resource in which case this function fails to work as intended, based on the review I could return true for both `if status not found and resource is intended object`, `if status is not found but resource isn't set for us to verify`.

Reply:
We don't need to complicate this so much, we know exactly the function that produced the error and the potential error space coming from it, Also we are looking for a subset of a specific error code that comes from a trusted source (the k8s client). With all of this in mind, you don't need to write code that is so defensive